lib/ansible/modules/extended_mount_facts.py

```diff
@@ -0,0 +1,298 @@
+# MacOSX: //user@hostname/public on /mnt/cifs (smbfs, nodev, nosuid, mounted by user)
+BSD_MOUNT_RE = re.compile(r"^(?P<device>\S+) on (?P<mount>\S+) \((?P<fstype>.+)\)$")
+
+# AIX: hostname /mount /mounted_over nfs Dec 17 08:06 ro, log  =/dev/hd8
+AIX_MOUNT_RE = re.compile(r"^(?P<node>\S+)\s+(?P<mount>\S+)\s+(?P<mounted_over>\S+)\s+(?P<fstype>\S+)\s+(?P<time>\S+\s+\d+\s+\d+:\d+)\s+(?P<options>.*)$")
+
+# TODO...?
+
+# No mount parsing preferably
+# Fedora, Debian, Ubuntu, Arch use /etc/mtab or /proc/mounts
+# Solaris uses /etc/mnttab
+
+
+def get_mount_info(
+    module: AnsibleModule,
+    mount: str,
+    device: str,
+    uuids: dict[str, str],
+    udevadm_uuid: Callable[[str], str]
+) -> tuple[Optional[dict], Optional[str]]:
+    """
+    Attempts to get the mount size and UUID within an optional timeout.
+    """
+    def _get_mount_info(mount, device, uuids, udevadm_uuid):
+        mount_size = get_mount_size(mount)
+        uuid = uuids.get(device, udevadm_uuid(device))
+        return mount_size, uuid
+
+    if module.params["timeout"] is None:
+        return _get_mount_info(mount, device, uuids, udevadm_uuid)
+
+    seconds = module.params["timeout"]
+    on_error = module.params["on_timeout"]
+    try:
+        return timeout.timeout(seconds)(_get_mount_info)(mount, device, uuids, udevadm_uuid)
+    except timeout.TimeoutError:
+        if on_error == "error":
+            module.fail_json(msg=f"{device} on {mount} timed out")
+        elif on_error == "warn":
+            module.warn(f"{device} on {mount} timed out")
+
+    return None, None
+
+
+def parse_mount(line: str) -> Optional[dict[str, str]]:
+    """
+    Parse a line of mount information and return a dictionary with mount details.
+    """
+    match = None
+    if platform.startswith(("freebsd", "netbsd", "openbsd", "dragonfly", "darwin", "linux")):
+        match = BSD_MOUNT_RE.match(line)
+    if platform.startswith(("aix", "linux")):
+        match = match or AIX_MOUNT_RE.match(line)
+    if match is None:
+        return None
+
+    mount_info = {"mount": match.group("mount")}
+    if len(match.groups()) == 3:
+        if "," in (fstype := match.group("fstype")):
+            fstype, options = fstype.split(", ", 1)
+        else:
+            options = ""
+        mount_info.update({
+            "fstype": fstype,
+            "options": options,
+            "device": match.group("device")
+        })
+    else:
+        device = match.group("node") + ":" + match.group("mounted_over")
+        mount_info.update({
+            "fstype": match.group("fstype"),
+            "options": match.group("options"),
+            "time": match.group("time"),
+            "device": device
+        })
+    return mount_info
+
+
+def parse_mount_output(module: AnsibleModule) -> list[dict[str, str]]:
+    """
+    Parse output from the mount command.
+    """
+    mount_path = module.get_bin_path("mount")
+    mount_output = ""
+    if mount_path:
```

Comment:
Let's bail out early ... 

```
mounts = []
if mount_path is None:
   return mounts
```

Reply:
I wanted to have 1 return statement, but yeah I just added a second...